src/main/java/org/shanerx/tradeshop/listeners/ShopProtectionListener.java

```diff
@@ -144,79 +142,53 @@ public void onEntityExplodeItem(EntityExplodeEvent event) {
 		event.blockList().removeAll(toRemove);
 	}
 
-    @EventHandler(priority = EventPriority.HIGHEST)
+    @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
     public void onBlockBreak(BlockBreakEvent event) {
         Player player = event.getPlayer();
         Block block = event.getBlock();
+        Shop shop = null;
 
         if (ShopType.isShop(block)) {
-            Shop shop = Shop.loadShop((Sign) block.getState());
+            shop = Shop.loadShop((Sign) block.getState());
             if (shop == null)
                 return;
-
             if (player.hasPermission(Permissions.ADMIN.getPerm()) || player.getUniqueId().equals(shop.getOwner().getUUID())) {
-	
-	            PlayerShopDestroyEvent destroyEvent = new PlayerShopDestroyEvent(player, shop);
-	            Bukkit.getPluginManager().callEvent(destroyEvent);
-	            if (destroyEvent.isCancelled()) {
-	            	event.setCancelled(true);
-		            return;
-	            }
-	            
-                if (shop.getStorage() != null) {
-                    new ShopChest(shop.getStorage().getLocation()).resetName();
+                PlayerShopDestroyEvent destroyEvent = new PlayerShopDestroyEvent(player, shop);
+                Bukkit.getPluginManager().callEvent(destroyEvent);
+                if (destroyEvent.isCancelled()) {
+                    event.setCancelled(true);
+                    return;
                 }
 
+                shop.getChestAsSC().resetName();
                 shop.remove();
                 return;
             }
+
             event.setCancelled(true);
             player.sendMessage(Message.NO_TS_DESTROY.getPrefixed());
 
-        } else if (plugin.getListManager().isInventory(block)) {
-            BlockState bs = block.getState();
-            if (!ShopChest.isShopChest(bs.getBlock()))
-                return;
-
-            if (player.hasPermission(Permissions.ADMIN.getPerm())) {
-                new ShopChest(block.getLocation()).resetName();
-                return;
-            }
-
-            Sign s = findShopSign(block);
-            if (s == null) {
-                new ShopChest(block.getLocation()).resetName();
-                return;
-            }
-
-            if (!ShopType.isShop(s)) {
-                new ShopChest(block.getLocation()).resetName();
+        } else if (ShopChest.isShopChest(block)) {
+            shop = new ShopChest(block.getLocation()).getShop();
+            if (shop == null)
```

Comment:
sleek

Reply:
this was added a bit ago, just didn't get it everywhere